File: Programmers/challenge7.py
def solution(absolutes, signs):
    answer = 0

    for i in range(len(absolutes)):
        if signs[i]:
            answer += absolutes[i]
        else:
            answer += absolutes[i] * -1

    return answer

absolutes = [4, 7, 12]
signs = [True, False, True]
# answer = 9
print(solution(absolutes, signs))

absolutes = [1, 2, 3]
signs = [False, False, True]
# answer = 0
print(solution(absolutes, signs))

File: Programmers/challenge7.js
const convertTernary = (n) => {
    let result = '';
    while(n > 0){
        result = (n % 3).toString() + result;
        n = Math.floor(n / 3);
    }
    return result;
}

const convertDecimal = (n, base) => {
    let result = 0;
    let power = 1;
    for(let i = n.length - 1; i >= 0; i--){
        result += n[i] * power;
        power *= base;
    }
    return result;
}

function solution(n) {
    var answer = convertDecimal(convertTernary(n).split("").reverse(), 3);
    return answer;
}